bip-0352.mediawiki

```diff
@@ -0,0 +1,471 @@
+
+Alice performs the tweak with the sum of her input private keys in the following manner:
+
+* Let ''outpoints_hash = hash(txid<sub>0</sub> || vout<sub>0</sub> || … txid<sub>n</sub> || vout<sub>n</sub>)''
+* Let ''a = a<sub>0</sub> + a<sub>1</sub> … + a<sub>n</sub>''
+* Let ''P<sub>0</sub> = B + hash(outpoints_hash·a·B || 0)·G''
+
+''' Spend and Scan Key '''
+
+Since Bob needs his private key ''b'' to check for incoming payments, this requires ''b'' to be exposed to an online device. To minimize the risks involved, Bob can instead publish an address of the form ''(B<sub>scan</sub>, B<sub>spend</sub>)''. This allows Bob to keep ''b<sub>spend</sub>'' in offline cold storage and perform the scanning with the public key ''B<sub>spend</sub>'' and private key ''b<sub>scan</sub>''. Alice performs the tweak using both of Bob's public keys in the following manner:
```

Comment:
There should be a footnote explaining the rationale for having an independent scan key, as opposed to deriving it from the spend key. You mentioned in the PR review club that one reason was redundancy (a compromised spend key doesn't lead to loss of funds without also a compromised scan key), which I take some issue with<sup>1</sup>, and also something about labels not working without an independent scan key, which sounds better, but would be great to have an explanation for.

<hr>

<sup>1</sup> This kind of silent redundancy (pun not intended) isn't IMO something that should be purposely built into a protocol as it breaks the separation of responsibilities. The scan key is responsible for scanning for payments, and the consequence of its compromise is loss of privacy. The spend key is responsible for spending received outputs, and the consequence of its compromise should be understood as loss of funds, even if by a quirk of the protocol it doesn't necessarily have to be. Redundancy can be built on top of the spend key by using MuSig2 or FROST, not by treating the scan key as more important than it is.

Also the fact that both the spend and scan key are required for spending should perhaps be explicitly mentioned somewhere. Due to the naming one might think that the spend key is enough.

Reply:
> labels not working without an independent scan key

To be precise, my point was that for labels to work, there is no way for the sender to infer the spend public key from the scan public key. The suggestion from the review club was if we can somehow derive the spend private key from the scan private key, there might be a way for the sender to derive the spend public key from the scan public key, allowing us to post one public key instead of two.

Even if the spend key is derived from the scan key, labels work by adding another public key (_m·G_) to the spend public key, where _m_ is unknown to the sender.

> This kind of silent redundancy (pun not intended) isn't IMO something that should be purposely built into a protocol as it breaks the separation of responsibilities.

You make a really good point. First, I agree we need to be more explicit: both the spend key and scan key are required to spend funds. The scan key is needed for finding the "shared secret" portion of the private key and the spend key is needed to ensure it is only spendable by the recipient. Given that, having independent derivation is a weak argument in that it only addresses one very narrow case: you compromise the spend key, but you don't compromise your BIP32 master key.

Originally, we had the scan key defined as the hash of the spend key, which then allows you to derive the spend key with or without BIP32. What I'm unsure of is what's easier/safer for hardware wallets: hashing a spend key and exporting that, or exporting a private key from a specific derivation path. Depending on the answer to that question, I may be more inclined to go back to the original: _scan_key = hash(spend_key)_

Reply:
>there might be a way for the sender to derive the spend public key from the scan public key, allowing us to post one public key instead of two

That is impossible (unless scan key == spend key, but then that would defeat the purpose)

Reply:
@RubenSomsen I think we may still want a footnote here? Based on recent discussions, our rational for how it is defined now is simplicity and likely the most compatible across wallet implementations today. That being said, I do think we will need a wallet BIP in the future for silent payments, as this BIP should stay primarily focused on just the protocol.

Reply:
I'd say no. There is a "Spending" section in the specification which specifies the full key needed for spending. I also think its better to define _how_ to derive the spend and scan key in a separate BIP, rather than add more footnotes here.